src/averageAges.js

```diff
@@ -55,7 +67,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childrenWithMother = people.filter(child =>
+    child.mother && people.some(parent => parent.name === child.mother)
+  );
+  const filteredChildrenWithMother = onlyWithSon
+    ? childrenWithMother.filter(child => child.sex === 'm')
+    : childrenWithMother;
+  const ageDifferences = filteredChildrenWithMother.map(person =>
+    person.born - people.find(parent => parent.name === person.mother).born
+  );
```

Comment:
I would rather move `parent.name === person.mother` to the variable to make the code easier